src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,80 @@
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String VALUE_CRITERIA = "supply";
+    private static final String LINE_SEPARATOR = " ";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        List<Integer> listOfSupplyCosts = new ArrayList<>();
+        List<Integer> listOfBuyCosts = new ArrayList<>();
+        String result = createReport(fromFileName, listOfSupplyCosts,
+                                     listOfBuyCosts, stringBuilder);
+        createAndWriteToFile(toFileName, result);
+    }
+
+    public static String createReport(String fromFile, List<Integer> supplies,
+                                      List<Integer> buys, StringBuilder stringBuilder) {
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile))) {
+            String string = bufferedReader.readLine();
+            String[] splitedString = null;
 
-    public void getStatistic(String fromFileName, String toFileName) {
+            while (string != null) {
+                splitedString = string.split(LINE_SEPARATOR);
+                for (String value : splitedString) {
+                    if (value.contains(VALUE_CRITERIA)) {
+                        supplies.add(Integer.valueOf(value
+                                            .substring(value.indexOf(",") + 1, value.length())));
+                    } else {
+                        buys.add(Integer.valueOf(value
+                                        .substring(value.indexOf(",") + 1, value.length())));
+                    }
+                }
+                string = bufferedReader.readLine();
+            }
+
+            int sumOfSupplies = 0;
+            for (int number : supplies) {
+                sumOfSupplies += number;
+            }
+
+            int sumOfBuys = 0;
+            for (int number : buys) {
+                sumOfBuys += number;
+            }
+            return stringBuilder.append("supply,").append(sumOfSupplies)
+                    .append(System.lineSeparator()).append("buy,")
+                    .append(sumOfBuys).append(System.lineSeparator())
+                    .append("result,").append(sumOfSupplies - sumOfBuys).toString();
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read data from file", e);
+        }
+    }
+
+    public static void createAndWriteToFile(String toFileName, String data) {
+        File resultFile = new File(toFileName);
+        try {
+            resultFile.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create file", e);
+        }
+        try {
+            Files.write(resultFile.toPath(), data.getBytes());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file", e);
+        }
+    }
```

Comment:
```suggestion
    public static void writeToFile(String toFileName, String data) {
        try {
            Files.write(new File(toFileName).toPath(), data.getBytes());
        } catch (IOException e) {
            throw new RuntimeException("Can't write data to file: " + toFileName, e);
        }
    }
```
no need to specifically create new file, `Files.write()` will create it for you, if it doesn't exists
add info about file to your exception, so it's more helpful